### PR TITLE
Running chart e2es

### PR DESCRIPTION
**Pull request by foxish**
Partially completes https://github.com/kubernetes/test-infra/issues/1887

cc @krzyzacy

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml [hunk 1]
```diff
@@ -76,6 +76,15 @@
         trigger-job: 'ci-kubernetes-build'
     # END AWS
 
+    # START CHARTS
+    - kubernetes-e2e-charts-gce: # foxish
+        job-name: ci-kubernetes-e2e-charts-gce
+        timeout: 100
```
Comment (krzyzacy) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml]: bootstrap timeout need to be longer than e2e timeout, which is > 180m. 200 would be fine.

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -865,7 +865,9 @@ test_groups:
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-latest-upgrade-cluster
 - name: kubernetes-e2e-gke-latest-upgrade-master
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-latest-upgrade-master
-
+# charts tests
+- name: ci-kubernetes-e2e-charts-gce
+  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-charts-gce
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: Oh, also add it to one of the dashboards, or create a new charts one ;p

**Comment by krzyzacy** [comment]
```
self.assertTrue(os.access(job_path, os.X_OK|os.R_OK), job_path)
AssertionError: /workspace/k8s.io/test-infra/jenkins/../jobs/ci-kubernetes-e2e-charts-gce.env
```
hummm... I'll update it to not require +X for env files.

File: testgrid/config/config.yaml (hunk 2 above)
Comment (foxish) [inline testgrid/config/config.yaml]: Thanks for the quick review :) Anything else I've missed? 
I don't think I can push a new image of e2e-runner. Is that something someone else can help me do after I've addressed all comments?

Comment (krzyzacy) [inline testgrid/config/config.yaml]: Let @fejta take a look at the kubetest changes :-)

**Comment by krzyzacy** [comment]
travis is flaky again :-(

File: kubetest/e2e.go [hunk 3]
```diff
@@ -483,6 +488,21 @@ func DumpClusterLogs(location string) error {
 	return finishRunning(exec.Command("./cluster/log-dump.sh", location))
 }
 
+func ChartsTest() error {
+	// Clone the charts repository
+	if err := finishRunning(exec.Command("git", "clone", "--depth", "50",
```
Comment (spxtr) [inline kubetest/e2e.go]: It would be great if `bootstrap.py` could handle this instead. @fejta can bootstrap clone multiple repos yet?

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

File: kubetest/e2e.go (hunk 3 above)
Comment (fejta) [inline kubetest/e2e.go]: What are the multiple repos we need?

**Comment by fejta** [comment]
What functionality of kubetest are you trying to use?

It seems like you:
a) want to use a bootstrap command that includes a --repo=kubernetes/charts flag, and
b) want a scenario that calls charts/test/e2e.sh.

**Comment by krzyzacy** [comment]
@fejta It wants to test charts against a cluster with latest kubernetes, which is currently done with kubetest --up

**Comment by fejta** [comment]
Ah, well in that case a) is still valid (send --repo flag to bootstrap) and I would like to eliminate calling docker run inside a docker container. Aka this should call charts/test/changed.sh directly

**Comment by foxish** [comment]
I just tried using the --repo flag with bootstrap. It appears that bootstrap checks the repository out to my local file system, and then launches a docker container "kubekins-e2e" that actually runs the kubetest binary. Do I need to do something special to make the locally checked out copy of the repo accessible within the docker container running kubetest? It is not very evident to me how I can use the two together.

**Comment by fejta** [comment]
Yes, you need to map in the repo into the container with the `-v $GOPATH/src/k8s.io/charts:/src/k8s.io/charts` which will result in the `$GOPATH/src/k8s.io/charts` dir on the host being available to the container at `/src/k8s.io/charts`. 

See any of the unit/verify jobs for examples of how this is done: https://github.com/kubernetes/test-infra/blob/master/scenarios/kubernetes_verify.py#L85

Will you add a --charts flag to the e2e scenario for this?

File: jobs/config.json [hunk 4]
```diff
@@ -948,6 +948,18 @@
   ]
 },
 
+"ci-kubernetes-e2e-charts-gce": {
+  "scenario": "kubernetes_e2e",
+  "args": [
+    "--env-file=platforms/gce.env",
+    "--env-file=jobs/ci-kubernetes-e2e-charts-gce.env",
+    "--test=false",
```
Comment (fejta) [inline jobs/config.json]: FYI you'll probably need a --tag arg to make sure you're using the right image?

Comment (foxish) [inline jobs/config.json]: I was expecting that we'd push a new image and make that the default.

Comment (fejta) [inline jobs/config.json]: That works too

Comment (foxish) [inline jobs/config.json]: I don't think I have privileges, can someone help me push a new image? I'm fixing the test issue now, but it should be all set after that. The file `charts/test/helm-test-e2e.sh` does not exist and the test will fail at first. Once I setup the tests in the charts repo, the tests should start running. Does that sound reasonable?

**Comment by foxish** [comment]
@spxtr @fejta All green. PTAL. Thanks!

**Comment by foxish** [comment]
Thanks @fejta 

I'll push the new image tomorrow morning and watch the canary for a while
If all is well, I'll update the PR to point to the new image and submit it.

**Comment by foxish** [comment]
Just resolved conflicts, pushing a new image. Will wait for it to be okay in the canary deployment.

/cc @krzyzacy

**Comment by foxish** [comment]
Tests passed on canary job. Updated the tag. Any final comments before merging?

**Comment by krzyzacy** [comment]
Give it a practical try :-) feel free to merge on green

**Comment by ixdy** [comment]
It seems like this job has never passed:
```
error starting /src/k8s.io/charts/test/helm-test-e2e.sh: fork/exec /src/k8s.io/charts/test/helm-test-e2e.sh: no such file or directory]
```

**Comment by foxish** [comment]
Yes. It is waiting for https://github.com/kubernetes/charts/pull/824

**Pull request merged**